src/device-state/current-state.ts

```diff
@@ -108,25 +108,25 @@ const getStateDiff = (): DeviceStatus => {
 	}
 
 	const diff = {
-		local: _(stateForReport.local)
-			.omitBy((val, key: keyof DeviceStatus['local']) =>
-				_.isEqual(lastReportedLocal[key], val),
-			)
-			.omit(INTERNAL_STATE_KEYS)
-			.value(),
-		dependent: _(stateForReport.dependent)
-			.omitBy((val, key: keyof DeviceStatus['dependent']) =>
+		local: _.omitBy(
+			stateForReport.local,
+			(val, key: keyof NonNullable<DeviceStatus['local']>) =>
+				INTERNAL_STATE_KEYS.includes(key) ||
```

Comment:
Moving this before the `_.isEqual` avoids the need to run `_.isEqual` when we're just going to discard it anyway, I think the keys we discard should be fairly cheap to compare so it probably won't matter much overall but it makes sense as a best practice to put the cheaper check first